src/main/java/core/basesyntax/JavaStreamApi.java

```diff
@@ -85,6 +105,10 @@ public Double averageSumOdd(List<Integer> numbers) {
      * Задача: вивести все имена кошек в которых хозяева это девушки старше 18 лет</p>
      **/
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(x -> x.getAge() >= femaleAge && x.getSex() == People.Sex.WOMEN)
+                .flatMap(x -> x.getCatList().stream())
+                .map(x -> x.getName())
```

Comment:
```suggestion
                .map(Cat::getName)
```
here we can use method reference

Reply:
done